libcaf_core/caf/json_reader.test.cpp

```diff
@@ -251,6 +252,22 @@ struct fixture {
     test_cases.emplace_back(std::move(f));
   }
 
+  // Specialization for `double` to add a test case for a given input.
+  void add_test_case(std::string_view input, double val) {
```

Comment:
I think we can just use `approx` in the template version. Comparing integers with `approx` without setting an `epsilon` compares for equality (since the default `epsilon` is 0).